Simulator/actions/simulator_actions.js

```diff
@@ -0,0 +1,101 @@
+export const setPhaseInProgress = createAction('SET_PHASE_IN_PROGRESS');
+export const setPhaseInitCallbackCalled = createAction('SET_PHASE_INIT_CALLBACK_CALLED');
+export const setPhaseSubmitCalled = createAction('SET_PHASE_SUBMIT_CALLED');
+
+// Window Actions
+export const setSimulatorWindow = createAction('SET_SIMULATOR_WINDOW');
+export const setShouldHaveGatherDataFrame = createAction('SET_SHOULD_HAVE_GATHER_DATA_FRAME');
+
+// Table actions
+export const setTables = createAction('SET_TABLES');
+export const addTables = createAction('ADD_TABLES');
+
+// Reset Actions
+export const resetState = createAction('RESET_STATE');
+export const resetPhaseState = createAction('RESET_PHASE_STATE');
+export const resetWdcAttrs = createAction('RESET_WDC_ATTRS');
+export const resetTables = createAction('RESET_TABLES');
+export const resetTableData = createAction('RESET_TABLE_Data');
+
+
+// Thunks (and Composed Actions)
+// More info can be found here:
+// https://github.com/gaearon/redux-thunk
+
+// Phase Control Thunks
+export function startConnector(phase) {
+  return (dispatch, getState) => {
+    // Commit url changes once we are sure the user is done
+    const { wdcUrl } = getState();
+    Cookie.set('lastUrl', wdcUrl);
+
+    // Clean up simulator and get ready for starting connector
+    dispatch(resetTables());
+    dispatch(setCurrentPhase(phase));
+    dispatch(setPhaseInProgress(true));
+    dispatch(closeSimulatorWindow());
+    dispatch(setWindowAsExternal());
+  };
+}
+
+export function startGatherDataPhase() {
+  return (dispatch) => {
+    // Start Data Gather Phase, close interactive window,
+    // open iframe
+    dispatch(setCurrentPhase(consts.phases.GATHER_DATA));
+    dispatch(setPhaseInProgress(true));
+    dispatch(closeSimulatorWindow());
+    dispatch(setShouldHaveGatherDataFrame(true));
+  };
+}
+
+// Simulator Window Thunks
+// Opens a window for the wdc and sets it as the window we will
+// be using for communication with the wdc. Used in the interactive and auth phases
+export function setWindowAsExternal() {
+  return (dispatch, getState) => {
+    const { wdcUrl } = getState();
+    const simulatorWindow = window.open(wdcUrl, 'simulator', consts.WINDOW_PROPS);
+    dispatch(setSimulatorWindow(simulatorWindow));
+  };
+}
+
+// Gets a ref to the iframe we open during the data gather phase
+// and set's its content window as the window we will be using to
+// communicate with the wdc. Acts as the headless browser used
+// in the desktop version of the connector lifecycle
+export function setWindowAsGatherFrame(iframe) {
+  return (dispatch) => {
+    if (!!iframe) {
+      dispatch(setSimulatorWindow(iframe.contentWindow));
+    }
+  };
+}
+
+export function closeSimulatorWindow() {
+  return (dispatch, getState) => {
+    const { simulatorWindow } = getState();
+    if (simulatorWindow) {
+      simulatorWindow.close();
+    }
+
+    dispatch(setShouldHaveGatherDataFrame(false));
```

Comment:
Right now state.simulatorWindow points to whatever window we are using to communicate to the wdc, which is either the new window opened in the interactive phase or the content window of the iframe we open in the gather data phase so when we close that window we want to signal to the simulator component that we don't want to immediately recreate the iframe if their was one. That code split is convenient, but I could see it being worth the extra verbosity to split the two objects into separate parts of the state to make it more clear.